@attr(tags=['advancedns', 'test'])
def test_04_network_restart_with_cleanup(self):
    'Test LB rules on Netscaler after network restart with cleanup\n        '
    self.debug(('Restarting the network: %s' % self.network.id))
    self.network.restart(self.apiclient, cleanup=True)
    self.debug(('SSH into netscaler: %s' % self.services['netscaler_VPX']['ipaddress']))
    try:
        network_list = Network.list(self.apiclient, id=self.network.id, listall=True)
        nw = network_list[0]
        ssh_client = SshClient(self.services['netscaler_VPX']['ipaddress'], self.services['netscaler_VPX']['port'], self.services['netscaler_VPX']['username'], self.services['netscaler_VPX']['password'])
        cmd = ('show vlan %s' % nw.vlan)
        self.debug(('command: %s' % cmd))
        res = ssh_client.execute(cmd)
        result = str(res)
        self.debug(('Output: %s' % result))
        self.assertNotEqual(result.count('ERROR: No such resource '), 1, 'Netscaler should have vlan configured for the network')
        cmd = ('show lb vserver Cloud-VirtualServer-%s-%s' % (self.public_ip.ipaddress.ipaddress, self.lb_rule.publicport))
        self.debug(('command: %s' % cmd))
        res = ssh_client.execute(cmd)
        result = str(res)
        self.debug(('Output: %s' % result))
        self.assertNotEqual(result.count('ERROR: No such resource'), 1, 'Netscaler should have vlan configured for the network')
    except Exception as e:
        self.fail(('SSH Access failed for %s: %s' % (self.services['netscaler_VPX']['ipaddress'], e)))
    return
